pkg/printer/printer.go

```diff
@@ -265,6 +267,99 @@ func (p *Printer) WriteProtoFlow(f v1.Flow) error {
 	return nil
 }
 
+// joinWithCutOff performs a strings.Join, but will omit elements if the
+// resulting string is longer than targetLen. The resulting string may be
+// longer than targetLen, as it will always print at least one element.
+// The number of omitted elements is appended to the resulting string as
+// " (and N more)".
+func joinWithCutOff(elems []string, sep string, targetLen int) string {
+	strLen := 0
+	end := len(elems)
+	for i, elem := range elems {
+		strLen += len(elem) + len(sep)
+		if strLen > targetLen && i > 0 {
+			end = i
+			break
+		}
+	}
+
+	joined := strings.Join(elems[:end], sep)
+	omitted := len(elems) - end
+	if omitted == 0 {
+		return joined
+	}
+
+	return fmt.Sprintf("%s (and %d more)", joined, omitted)
+}
+
+// WriteProtoNodeStatusEvent writes a node status event into the error stream
+func (p *Printer) WriteProtoNodeStatusEvent(r *observerpb.GetFlowsResponse) error {
+	s := r.GetNodeStatus()
+	if s == nil {
+		return errors.New("not a node status event")
+	}
+
+	if !p.opts.enableDebug {
+		switch s.GetStateChange() {
+		case relaypb.NodeState_NODE_ERROR, relaypb.NodeState_NODE_UNAVAILABLE:
+			break
+		default:
+			// skips informal messages in non-debug mode
+			return nil
+		}
+	}
+
+	switch p.opts.output {
+	case JSONOutput:
+		return json.NewEncoder(p.opts.werr).Encode(r)
+	case DictOutput:
+		// this is a bit crude, but in case stdout and stderr are interleaved,
+		// we want to make sure the separators are still printed to clearly
+		// separate flows from node events.
+		if p.line != 0 {
+			_, err := fmt.Fprintln(p.opts.w, "------------")
```

Comment:
share this const maybe? it's technically the length of the longest key, so may differ based on the message